ocs_ci/deployment/provider_client/storage_client_deployment.py

```diff
@@ -0,0 +1,630 @@
+            assert (
+                self.rados_utils.check_phase_of_rados_namespace()
+            ), "The radosnamespace is not in Ready phase"
+
+            # Validate storageclassrequests created
+            storage_class_classes = get_all_storageclass_names()
+            for storage_class in self.storage_class_claims:
+                assert (
+                    storage_class in storage_class_classes
+                ), "Storage classes ae not created as expected"
+
+        else:
+            # Create ODF subscription for storage-client
+            self.odf_installation_on_client()
+
+            # Fetch storage provider endpoint details
+            storage_provider_endpoint = self.ocp_obj.exec_oc_cmd(
+                (
+                    f"get storageclusters.ocs.openshift.io -n {config.ENV_DATA['cluster_namespace']}"
+                    + " -o jsonpath={'.items[*].status.storageProviderEndpoint'}"
+                ),
+                out_yaml_format=False,
+            )
+            log.info(f"storage provider endpoint is: {storage_provider_endpoint}")
+
+            # Create Network Policy
+            self.create_network_policy(
+                namespace_to_create_storage_client=constants.OPENSHIFT_STORAGE_CLIENT_NAMESPACE
+            )
+            onboarding_token = self.onboarding_token_generation_from_ui()
+
+            # Create native storage client
+            create_storage_client(
+                storage_provider_endpoint=storage_provider_endpoint,
+                onboarding_token=onboarding_token,
+            )
+
+        # Check nooba db pod is up and running
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.NOOBAA_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.RGW_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.PROVIDER_SERVER_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        self.pod_obj.wait_for_resource(
+            condition=constants.STATUS_RUNNING,
+            selector=constants.RGW_APP_LABEL,
+            resource_count=1,
+            timeout=300,
+        )
+        list_of_rgw_pods = pod.get_rgw_pods(
+            namespace=config.ENV_DATA["cluster_namespace"]
+        )
+        rgw_pod_obj = list_of_rgw_pods[0]
+        restart_count_for_rgw_pod = pod.get_pod_restarts_count(
+            list_of_pods=list_of_rgw_pods,
+            namespace=config.ENV_DATA["cluster_namespace"],
+        )
+        rgw_pod_restart_count = restart_count_for_rgw_pod[rgw_pod_obj.name]
+        log.info(f"restart count for rgw pod is: {rgw_pod_restart_count}")
+        assert (
```

Comment:
please consider to move this assertion to separate 'verification' method.
btw this  'verification' method will be handy as a post-setup verification upon any test we do.

Reply:
Have separated verification method.